common/common/src/main/java/io/helidon/common/Default.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A container class for default values related types for Helidon declarative.
+ *
+ * @deprecated this API is part of incubating features of Helidon. This API may change including backward incompatible changes
+ *               and full removal. We welcome feedback for incubating features.
+ */
+@Deprecated
```

Comment:
It's unfortunate there isn't a more appropriate annotation for incubating stuff. Should we create one?

Reply:
We do have a `Features.Incubating`, but that is meant for `module-info.java`.
I would not have a problem reusing it, but the `@Deprecated` will shine in user's code and will be reported by compiler - this makes it really easy to see that you are doing something unusual (and it can be suppressed).

I do not know how to do anything similar unless we have an IDE plugin and an annotation processor

Reply:
The fact that it is reported as deprecated by the compiler is what I dislike the most being a new, incubating feature. But I understand we don't have a better option at this time.